scripts/migrations/062_update_packagIds_to_map.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+import couchdb
+import json
+import time
+import sys
+from datetime import datetime
+
+# ---------------------------------------
+# constants
+# ---------------------------------------
+
+DRY_RUN = False
+COUCHSERVER = "http://localhost:5984/"
+DBNAME = 'sw360db'
+BATCH_SIZE = 1000
+
+# ---------------------------------------
+# database connection
+# ---------------------------------------
+
+try:
+    couch = couchdb.Server(COUCHSERVER)
+    db = couch[DBNAME]
+except Exception as e:
+    print(f"Error connecting to CouchDB: {e}")
+    sys.exit(1)
+
+
+# ---------------------------------------
+# functions
+# ---------------------------------------
+
+def create_log_entry(project, status, error=None):
+    return {
+        "timestamp": datetime.now().isoformat(),
+        "status": status,
+        "error": str(error) if error else None,
+    }
+
+def update_package_ids_structure():
+    log = {
+        "migrationDate": datetime.now().isoformat(),
+        "dryRun": DRY_RUN,
+        "statistics": {
+            "totalProcessed": 0,
+            "successCount": 0,
+            "errorCount": 0
+        },
+        "entries": [],
+        "project_details": [],
+        "project_id_name": []
+    }
+
+    try:
+        print('Starting batch processing of projects with packageIds...')
+
+        # Initial query to get total count
+        count_query = {
+            "selector": {
+                "type": {"$eq": "project"},
+                "packageIds": {"$exists": True}
+            }
+        }
+        total_projects = len(list(db.find(count_query)))
+        print(f'Found {total_projects} projects to process')
+
+        skip = 0
+        while True:
+            # Query for current batch
+            batch_query = {
+                "selector": {
+                    "type": {"$eq": "project"},
+                    "packageIds": {"$exists": True}
+                },
+                "limit": BATCH_SIZE,
+                "skip": skip
+            }
+
+            projects = list(db.find(batch_query))
+            if not projects:
+                break
```

Comment:
Can we already reuse the list from previous call instead to fetch the projects at each iteration as an optimization?